### PR TITLE
Event Grid - update resource API versions and create the Bicep file

### PR DESCRIPTION
**Pull request by mumian**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

**Comment by azure-quickstarts** [comment]
@spelluru - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

File: quickstarts/microsoft.eventgrid/event-grid-subscription-and-storage/README.md [hunk 1]
```diff
@@ -9,6 +9,8 @@
 ![Best Practice Check](https://azurequickstartsservice.blob.core.windows.net/badges/quickstarts/microsoft.eventgrid/event-grid-subscription-and-storage/BestPracticeResult.svg)
 ![Cred Scan Check](https://azurequickstartsservice.blob.core.windows.net/badges/quickstarts/microsoft.eventgrid/event-grid-subscription-and-storage/CredScanResult.svg)
```
Comment (fran38) [inline quickstarts/microsoft.eventgrid/event-grid-subscription-and-storage/README.md:10]: - [ ] []()

File: quickstarts/microsoft.eventgrid/event-grid-subscription-and-storage/README.md [hunk 2]
```diff
@@ -9,6 +9,8 @@
...
 
+![Bicep Version](https://azurequickstartsservice.blob.core.windows.net/badges/quickstarts/microsoft.eventgrid/event-grid-subscription-and-storage/BicepVersion.svg)
```
Comment (fran38) [inline quickstarts/microsoft.eventgrid/event-grid-subscription-and-storage/README.md:12]: - []()```

```

File: quickstarts/microsoft.eventgrid/event-grid-subscription-and-storage/README.md [hunk 3]
```diff
@@ -9,6 +9,8 @@
...
+
 [![Deploy To Azure](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazure.svg?sanitize=true)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fquickstarts%2Fmicrosoft.eventgrid%2Fevent-grid-subscription-and-storage%2Fazuredeploy.json)
```
Comment (fran38) [inline quickstarts/microsoft.eventgrid/event-grid-subscription-and-storage/README.md:14]: - [ ]

File: quickstarts/microsoft.eventgrid/event-grid-subscription-and-storage/azuredeploy.json [hunk 4]
```diff
@@ -40,8 +47,8 @@
   "resources": [
     {
       "type": "Microsoft.Storage/storageAccounts",
-      "apiVersion": "2019-06-01",
-      "name": "[parameters('storageName')]",
+      "apiVersion": "2021-08-01",
+      "name": "[parameters('storageAccountName')]",
```
Comment (fran38) [inline quickstarts/microsoft.eventgrid/event-grid-subscription-and-storage/azuredeploy.json:51]: []()

File: quickstarts/microsoft.eventgrid/event-grid-subscription-and-storage/metadata.json [hunk 5]
```diff
@@ -6,5 +6,5 @@
   "summary": "This template creates Azure Blob Storage account and then creates an Event Grid subscription to that Blob.",
   "githubUsername": "spelluru",
   "docOwner": "spelluru",
-  "dateUpdated": "2021-04-26"
```
Comment (fran38) [inline quickstarts/microsoft.eventgrid/event-grid-subscription-and-storage/metadata.json:9]: Update~~

**Review by fran38 — APPROVED** [review]

**Comment by mumian** [comment]
@bmoore-msft - Can you merge this PR? Thanks.

**Comment by azure-quickstarts** [comment]
@spelluru - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Pull request merged**